### PR TITLE
Use Deployment for traces quickstart

### PR DESCRIPTION
**Pull request by hjet**
#### PR Description 

This PR moves quickstart deployment and configuration instructions into the Cloud docs. In addition, the trace collector Agent is rolled out as a Deployment instead of a DaemonSet by default. Tested with the Grafana [demo app](https://github.com/grafana/tns) and Grafana Cloud.

#### Notes to the Reviewer

Broken link pending https://github.com/grafana/website/pull/5340 

#### PR Checklist

- [ ] CHANGELOG updated 
- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: production/tanka/grafana-agent/v1/lib/deployment.libsonnet [hunk 1]
```diff
@@ -26,25 +31,61 @@ local agent = import '../internal/agent.libsonnet';
...
     agent:
       agent.newAgent(name, namespace, self._images.agent, self.config, use_daemonset=false) +
-      agent.withConfigHash(self._config_hash),
+      agent.withConfigHash(self._config_hash) + (
+        if has_tempo_config
+        then 
+          {
+            container+:: container.withEnvMixin([
+              k.core.v1.envVar.fromFieldPath('HOSTNAME', 'spec.nodeName'),
+            ]),
```
Comment (rfratto) [inline production/tanka/grafana-agent/v1/lib/deployment.libsonnet:73]: Hmm, I kind of think we should have this regardless if tempo configs are defined. It's necessary for so much stuff (logs, integrations, etc)

**Review by rfratto — COMMENTED** [review]
Cool, thanks. Do we document sampling strategies anywhere in the tempo quickstart?

File: production/tanka/grafana-agent/v1/lib/deployment.libsonnet (hunk 1 above)
Comment (hjet) [reply production/tanka/grafana-agent/v1/lib/deployment.libsonnet]: Good call, moved it out of the `if` clause. Note that this adds it to `agent-bare.yaml` as well (which doesn't really matter)

**Comment by hjet** [comment]
> Cool, thanks. Do we document sampling strategies anywhere in the tempo quickstart?

We don't, but I'll make a docs PR briefly addressing them and also link out to the Tempo docs

**Review by rfratto — APPROVED** [review]
This is great, thanks!

**Pull request merged**